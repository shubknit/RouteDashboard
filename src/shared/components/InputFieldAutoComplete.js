import React, { Component } from 'react';
import { googleMapsLoad } from '../../services/googlemap/googleMapAPI';

export class InputFieldAutoComplete extends Component {
    inputAutoComplete;
    inputElement;

    constructor(){
        super();
    }

    componentDidMount(){
        this.renderAutoComplete();
    }

    handleOnChange = (e) => {
        const userInput = e.target.value.trim();
        this.props.handleOnSelectAddress(userInput, this.props.inputName);
    }

    renderAutoComplete = async () => {
        const maps = await this.props.googleMapsLoad();
        const locationDetails = {};
        this.inputAutoComplete = new maps.places.Autocomplete(this.inputElement);
        this.inputAutoComplete.addListener('place_changed', ()=>{
            const place = this.inputAutoComplete.getPlace();
            if (place && place.geometry) {
                locationDetails.lat = place.geometry.location.lat();
                locationDetails.lng = place.geometry.location.lng();
                this.props.handleOnSelectAddress(locationDetails, this.props.inputName)
            }
        })
    }

    render(){
        return (
            <div>
                <label> {this.props.label} </label>
                <input className = 'text-input'
                    type="text" 
                    name={this.props.inputName}
                    ref={ elem => this.inputElement = elem }
                    onChange={this.handleOnChange}
                />
            </div>
        )
    }
}

InputFieldAutoComplete.defaultProps = {
    googleMapsLoad
}    
