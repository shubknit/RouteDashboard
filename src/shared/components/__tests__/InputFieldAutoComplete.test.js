import React from 'react';
import { InputFieldAutoComplete } from '../InputFieldAutoComplete';
import renderer from 'react-test-renderer';


it('should render the InputFieldAutoComplete  correctly', () => {
    const wrapper = renderer.create(<InputFieldAutoComplete label = 'start' name = 'start'/>).toJSON();
    expect(wrapper).toMatchSnapshot();
})