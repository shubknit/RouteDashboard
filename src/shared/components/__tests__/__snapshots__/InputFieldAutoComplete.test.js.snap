// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the InputFieldAutoComplete  correctly 1`] = `
<div>
  <label>
     
    start
     
  </label>
  <input
    className="text-input"
    onChange={[Function]}
    type="text"
  />
</div>
`;
